lib/ansible/modules/dnf.py

```diff
@@ -317,6 +317,7 @@
   ansible.builtin.dnf:
     name: "*"
     state: latest
+    update_only: true
```

Comment:
This would prevent any dependencies introduced in new versions of packages not being installed and ultimately failing the upgrade. I believe wanting to upgrade all packages regardless of whether they pull in new packages as their dependencies is a valid use case that is worth documenting.